lib/go/Makefile

```diff
@@ -0,0 +1,68 @@
+all: build
+
+include ../../protoc.mk
+
+# If GOPATH isn't defined then set its default location.
+ifeq (,$(strip $(GOPATH)))
+GOPATH := $(HOME)/go
+else
+# If GOPATH is arleady set then update GOPATH to be its own
+# first element.
+GOPATH := $(word 1,$(subst :, ,$(GOPATH)))
+endif
+export GOPATH
+
+# Update PATH with the GOPATH/bin directory.
+PATH := $(GOPATH)/bin:$(PATH)
+export PATH
+
+# This is the recipe for getting and installing the go plug-in
+# for protoc
+PROTOC_GEN_GO := $(GOPATH)/bin/protoc-gen-go
+$(PROTOC_GEN_GO):
+	go get github.com/golang/protobuf/protoc-gen-go && \
+	go install github.com/golang/protobuf/protoc-gen-go
+
+# This is a list of the go packages required to generate
+# the go bindings for the csi specification.
+PROTOC_GO_PKGS += google.golang.org/grpc
+PROTOC_GO_PKGS += github.com/golang/protobuf/proto
+
+# The following recipe downloads the go packages necessary
+# to generate the go language bindings.
+PROTOC_GO_PKGS := $(addprefix $(GOPATH)/src/,$(PROTOC_GO_PKGS))
+$(PROTOC_GO_PKGS):
+	go get $(@:$(GOPATH)/src/%=%)
+
+PROTOC_GO := $(PROTOC_GO_PKGS) $(PROTOC_GEN_GO)
+protoc-go: $(PROTOC_GO)
+
+CSI_GO := csi.pb.go
+CSI_PROTO := ../../csi.proto
+CSI_GO_TMP := .build/csi.pb.go
+
+# This recipe generates the go language bindings to a temp area.
+$(CSI_GO_TMP): $(CSI_PROTO)
+	mkdir -p "$(@D)" && \
+	$(PROTOC) -I "$(<D)" --go_out=plugins=grpc:"$(@D)" "$<"
+
+# The temp language bindings are compared to the ones that are
+# versioned. If they are different then it means the language
+# bindings were not updated prior to being committed.
+$(CSI_GO): $(CSI_GO_TMP) | $(PROTOC_GO)
+ifeq (true,$(TRAVIS))
+	diff "$@" "$?"
+else
+	diff "$@" "$?" > /dev/null 2>&1 || cp -f "$?" "$@"
```

Comment:
So this output will depends on the version of protoc as well as the plugins i think? I am not sure if there will be any compatibility issue or not. For instance, what if a downstream wants to use a specific version of protoc because of some other dependencies?